File: pset2/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int get_word_count(string s);
int get_letter_count(string s);
int get_sentence_count(string s);
int get_index(int wc, int lc, int sc);


int main(void)
{
    string str = get_string("Enter a string plz: ");
    int lc = get_letter_count(str);
    int wc = get_word_count(str);
    int sc = get_sentence_count(str);
    int index = get_index(wc,lc,sc);
    if(index < 1)
        printf("Before Grade 1\n");
    else if(index > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", index);
}

int get_word_count(string s)
{
    int count = 0;
    int length = strlen(s);
    for(int i = 0; i < length - 1; i++)
    {
        if((count == 0 && isalpha(s[i])) || (isspace(s[i]) && isalpha(s[i+1])))
        {
            count++;
        }
    }
    return count;
}
int get_letter_count(string s)
{
    int count = 0;
    int length = strlen(s);
    for(int i = 0; i < length; i++)
    {
        if(isalpha(s[i]))
        {
            count++;
        }
    }
    return count;
}
int get_sentence_count(string s)
{
    int count = 0;
    int length = strlen(s);
    for(int i = 0; i < length; i++)
    {
        if(s[i] == '?' || s[i] == '!' || s[i] == '.' )
        {
            count++;
        }
    }
    return count;
}
int get_index(int wc, int lc, int sc)
{

    float multiplier = (float) 100 / (float)wc;
    float lc_per_wc = lc * multiplier;
    float sc_per_wc = sc * multiplier;
    float float_index = 0.0588 * lc_per_wc - 0.296 * sc_per_wc - 15.8;
    float rounded_index = round(float_index);
    return (int)rounded_index;
}

File: pset6/dna/readability.py
def main():
    text = input("Enter some text")
    letter_count = get_letter_count(text)
    word_count = get_word_count(text)
    sentence_count = get_sentence_count(text)
    index = get_index(word_count, letter_count, sentence_count)
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def get_word_count(text):
    count = 0
    text = text.split(" ")
    return len(text)


def get_letter_count(text):
    count = 0
    for char in text:
        if char.isalnum():
            count += 1
    return count


def get_sentence_count(text):
    count = 0
    for char in text:
        if char == "?" or char == "." or char == "!":
            count += 1
    return count


def get_index(word_count, letter_count, sentence_count):
    multiplier = 100 / word_count
    letter_count_per_word_count = letter_count * multiplier
    sentence_count_per_word_count = sentence_count * multiplier
    float_index = 0.0588 * letter_count_per_word_count - \
        0.296 * sentence_count_per_word_count - 15.8
    rounded_index = round(float_index)
    return rounded_index


main()